### ISSUE TITLE
Navigation bar not displayed when scanner presented modally

### ISSUE THREAD

**Issue by kjakm** [hyperoslo/BarcodeScanner#39]
I have been successfully using the scanner by pushing it on to the navigation stack. Now however I would like to present it. Strangely the built-in navigation bar is hidden. It seems like the `isBeingPresented` check is not returning the correct result.  The navigation bar does show but once the modal animation has completed and the bar has reached the top of the screen it disappears. I have pasted my code below.

```
        // Configure barcode scanner
        let controller = BarcodeScannerController()
        controller.codeDelegate = self
        controller.dismissalDelegate = self
        controller.errorDelegate = self
        controller.hidesBottomBarWhenPushed = true
                
        // Set strings
        BarcodeScanner.Info.loadingText = NSLocalizedString("Retrieving code...", comment: "")
            
        // Display scanner
        navigationController?.present(controller, animated: true, completion: nil)

```

Is there a way I can manually set the navigation bar visibility?

**Comment by festrs** [comment]
Same problem here!

**Comment by vadymmarkov** [comment]
@kjakm Have you tried to present navigation controller instead? 
```swift
let controller = BarcodeScannerController()
navigationController?.present(UINavigationController(rootViewController: controller), animated: true, completion: nil)
```

**Comment by ptaking** [comment]
@vadymmarkov It doesn't work either.

**Comment by jdbarbosa** [comment]
can't we remove the `isBeingPresented` check on `setupFrame` ?

**Comment by deanWombourne** [comment]
Hi - we are having the same issue in our app, has there been any progress on this? Thanks!

**Comment by deanWombourne** [comment]
Oh, this is fixed in the latest version (`3.0.0`), please can we close this issue?

**Comment by vadymmarkov** [comment]
I'm glad to hear the problem is solved 👍
